### PR TITLE
bkcrack: initial commit

### LINKED ISSUE #3122: [Tool Request] bkcrack
- **Name**: bkcrack
- **Source URL**: https://github.com/kimci86/bkcrack
- **Description**: Crack legacy zip encryption with Biham and Kocher's known plaintext attack.
- **AUR**: https://aur.archlinux.org/packages/bkcrack-git/

### PR DESCRIPTION
**Pull request by D4imyo**
This PR fixes #3122

totally not sure about this PKGBUILD lol

### CODE REVIEW

File: packages/bkcrack/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  install -dm 755 "$pkgdir/usr/share/$pkgname/"
+
+  install -Dm 755 "src/bkcrack" "$pkgdir/usr/bin/bkcrack"
+
+  install -Dm 644 license.txt "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
+  install -Dm 644 -t "$pkgdir/usr/share/doc/$pkgname/" *.md
+
+  rm license.txt *.md
+
+  cp -a * "$pkgdir/usr/share/$pkgname"
```
Comment (pwnfoo) [inline packages/bkcrack/PKGBUILD:43]: This copies the entire source code to `/usr/share/bkcrack` and is not required.

File: packages/bkcrack/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+pkgver() {
+  cd "$pkgname"
+
+  git describe --long --tags | sed 's/\([^-]*-g\)/r\1/;s/-/./g'
+}
+
+build() {
+  cd "$pkgname"
+
+  cmake .
```
Comment (pwnfoo) [inline packages/bkcrack/PKGBUILD:26]: `--config Release` is [recommended by upstream](https://github.com/kimci86/bkcrack#compile-from-source)

**Comment by pwnfoo** [comment]
@D4imyo Thanks a lot for the PR. 

It's best to add the tool name to the PR to avoid confusion. Inorder to make sure it closes the matching issue, you can just mention it in the PR body like : "This PR fixes #3122" :)

**Comment by noraj** [comment]
> totally not sure about this PKGBUILD lol

Please test it with this tool: https://gist.github.com/noraj/5045be32490f1161161e16fc5820066c#testing

File: packages/bkcrack/PKGBUILD (hunk 2 above)
Comment (D4imyo) [reply packages/bkcrack/PKGBUILD:26]: in order to use --config Release i will have to first generate the build directory and then the compile process,
and my code would be like this :
`
cmake -S . -B build -DCMAKE_INSTALL_PREFIX=install
cmake --build build --config Release
cmake --build build --config Release --target install
`
should i just copy these commands into the build function ? sorry new to cmake

**Comment by D4imyo** [comment]
> > totally not sure about this PKGBUILD lol
> 
> Please test it with this tool: https://gist.github.com/noraj/5045be32490f1161161e16fc5820066c#testing

it's working fine, i am just not sure about the build function

**Comment by noptrix** [comment]
@D4imyo thanks for thiss. please check my changes afterwards

**Pull request merged**